rcgen/src/lib.rs

```diff
@@ -114,8 +110,9 @@ println!("{}", key_pair.serialize_pem());
 )]
 pub fn generate_simple_self_signed(
 	subject_alt_names: impl Into<Vec<String>>,
-) -> Result<CertifiedKey, Error> {
-	Certificate::generate_self_signed(CertificateParams::new(subject_alt_names))
+	key_pair: &KeyPair,
+) -> Result<Certificate, Error> {
```

Comment:
If we eliminate the `CertifiedKey` type, maybe it would be better to return a tuple here? Then most people would save one line (plus they won't have to import the `KeyPair` type).